stackrox/stackrox
# ROX-22720: Add compliance operator installed flag

## Description

This PR adds a flag to mark whether the compliance operator is installed or not. This can be used by the UI to mark CO integrations as disabled/enabled.

Also includes a minor refactoring to reduce some if-branches to get the compliance operator deployment.

## Checklist
- [] ~~Evaluated and added CHANGELOG entry if required~~

If any of these don't apply, please comment below.

## Testing Performed

 - Install StackRox (central+ sensor, via `./deploy/...` scripts)
- Query the compliance integrations API, `operatorInstalled: false` should be set. 
```
$ roxcurl /v2/compliance/integrations
{"integrations":[{"id":"9e97b164-3177-4afc-b1bf-13b9675a1db5","version":"","clusterId":"49abcaaa-4260-417e-8568-b78221a4b5cd","clusterName":"remote","namespace":"","statusErrors":["deployment compliance-operator not found in any namespace"],"operatorInstalled":false}]}
```
- Clone https://github.com/stackrox/fake-compliance-operator
- Run `./apply.sh init`
- Run `roxcurl /v2/compliance/integrations` again, should be `operatorInstalled: true` now in the response
